erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -534,6 +539,65 @@ def update_billed_amount_based_on_po(po_detail, update_modified=True):
 
 	return updated_pr
 
+
+def update_billed_qty_based_on_po(po_detail, update_modified=True):
+	po_billed_qty = frappe.db.sql("""
+		SELECT
+			SUM(qty)
+		FROM
+			`tabPurchase Invoice Item`
+		WHERE
+			po_detail = %s
+				AND (pr_detail IS NULL OR pr_detail = '')
+				AND docstatus = 1
+	""", po_detail)
+
+	po_billed_qty = po_billed_qty and po_billed_qty[0][0] or 0
+
+	# Get all Delivery Note Item rows against the Purchase Order Item row
+	pr_details = frappe.db.sql("""
+		SELECT
+			pr_item.name,
+			pr_item.qty,
+			pr_item.parent
+		FROM
+			`tabPurchase Receipt Item` pr_item,
+			`tabPurchase Receipt` pr
+		WHERE
+			pr.name = pr_item.parent
+				AND pr_item.purchase_order_item = %s
+				AND pr.docstatus = 1
+				AND pr.is_return = 0
+		ORDER BY
+			pr.posting_date asc,
+			pr.posting_time asc,
+			pr.name asc
+	""", po_detail, as_dict=1)
```

Comment:
- Use ORM

Reply:
I'm not sure how I can do joins using the ORM. I tried using the list-filters way, but it didn't give me any results.